packages/fxa-settings/src/pages/Signup/container.tsx

```diff
@@ -75,22 +99,76 @@ const SignupContainer = ({
             hardNavigateToContentServer(
               `/signin?email=${queryParamModel.email}`
             );
+            // TODO: Probably move this to the Index page onsubmit once
+            // the index page is converted to React, we need to run it in
+            // signup and signin for Sync
+          } else if (isSyncWebChannel) {
+            firefox.fxaCanLinkAccount({ email: queryParamModel.email });
           }
+        } else if (isSyncWebChannel) {
+          // TODO: Probably move this to the Index page onsubmit once
+          // the index page is converted to React, we need to run it in
+          // signup and signin for Sync
+          firefox.fxaCanLinkAccount({ email: queryParamModel.email });
         }
       }
       setShowLoadingSpinner(false);
     })();
   });
 
+  useEffect(() => {
+    // This sends a webchannel message to the browser to prompt a response
+    // that we listen for.
+    // TODO: In content-server, we send this on app-start for all integration types.
+    // Do we want to move this somewhere else once the index page is Reactified?
+    if (isSyncWebChannel) {
+      firefox.addEventListener(
+        FirefoxCommand.FxAStatus,
+        handleFxAStatusSyncEngineEvent
+      );
+
+      // requestAnimationFrame ensures the event listener is added first
+      // otherwise, there is a race condition
+      requestAnimationFrame(() =>
```

Comment:
TIL

Reply:
I tried so many different things to get this work, and placing the listener and sender in the same `useEffect` with `setTimeOut` to `0` or `requestAnimationFrame` were the only things that worked. 😅 This is also good for animations for future reference (hence the name lol).